app/models/series.rb

```diff
@@ -190,16 +190,20 @@ def remove_author(author_to_remove)
   end
 
   # returns list of fandoms on this series
-  def allfandoms
-    works.collect(&:fandoms).flatten.compact.uniq.sort
+  def fandoms
+    tag_groups["Fandom"].sort
   end
 
   def author_tags
     self.work_tags.select{|t| t.type == "Relationship"}.sort + self.work_tags.select{|t| t.type == "Character"}.sort + self.work_tags.select{|t| t.type == "Freeform"}.sort
   end
 
   def tag_groups
-    self.work_tags.group_by { |t| t.type.to_s }
+    @tag_groups ||= if User.current_user.present?
+                      self.work_tags.group_by { |t| t.type.to_s }
+                    else
+                      self.work_tags.where(works: { restricted: false }).group_by { |t| t.type.to_s }
```

Comment:
I think this came up elsewhere, but should this also have `hidden_by_admin: false`?